app/src/main/java/com/cortado/minionify/preference/PreferencesSingleton.java

```diff
@@ -0,0 +1,46 @@
+package com.cortado.minionify.preference;
+
+import android.content.SharedPreferences;
+
+import com.cortado.minionify.mode.AppMode;
+import com.cortado.minionify.mode.exceptions.ModeNotExistsException;
+import com.cortado.minionify.preference.exceptions.PreferenceOperationFailure;
+
+public class PreferencesSingleton {
+    private static PreferencesSingleton INSTANCE = null;
+    private SharedPreferences mPreferences;
+
+    private static final String MODE_PREFERENCE_KEY = "mode";
+
+    private PreferencesSingleton(SharedPreferences prefs) {
+        mPreferences = prefs;
+    }
+
+    public static PreferencesSingleton getInstance(SharedPreferences prefs) {
+        if (INSTANCE == null) {
+            INSTANCE = new PreferencesSingleton(prefs);
+        }
+
+        return INSTANCE;
+    }
+
+    public void setAppMode(AppMode chosenAppMode) throws PreferenceOperationFailure {
```

Comment:
why is this method inside the singleton?

Reply:
Apparently I misunderstood the purpose of Singleton, read a little bit more about it, understood my mistake.